Changelog for v2.214.0. (#4379)
vertexproject/synapse: CHANGELOG.rst

Synapse Changelog
*****************

v2.214.0 - 2025-07-02
=====================

Automatic Migrations
--------------------
- Removed AuthGates for Queues which no longer exist in the Cortex.
  (`#4370 <https://github.com/vertexproject/synapse/pull/4370>`_)
- See :ref:`datamigration` for more information about automatic migrations.

Model Changes
-------------
- Added ``inet:service:instance:app`` to capture the app containing the
  instance.
  (`#4365 <https://github.com/vertexproject/synapse/pull/4365>`_)
- Added ``:app`` to the ``inet:service:object`` interface to capture the app
  containing the object.
  (`#4365 <https://github.com/vertexproject/synapse/pull/4365>`_)
- See :ref:`userguide_model_v2_214_0` for more detailed model changes.

Features and Enhancements
-------------------------
- Added ``synapse.tools.cortex.layer.dump`` and
  ``synapse.tools.cortex.layer.load`` command line tools for exporting node

>   edits from one layer and importing them to a layer on a different cortex.

Reviewer: ```suggestion
  edits from one layer and importing them to a layer on a different Cortex.
```

  (`#4325 <https://github.com/vertexproject/synapse/pull/4325>`_)
- Added a tool, ``synapse.tools.pkgs.gendocs``, for converting Storm package
  documentation from RST to Markdown.
  (`#4331 <https://github.com/vertexproject/synapse/pull/4331>`_)
- Added ``synapse.tools.axon.dump`` and ``synapse.tools.axon.load`` command
  line tools for exporting blobs from one Axon and importing them to another.
  (`#4333 <https://github.com/vertexproject/synapse/pull/4333>`_)
- Updated Cortex Storm pool behavior to prefer all remote pool members before
  the local leader.
  (`#4336 <https://github.com/vertexproject/synapse/pull/4336>`_)
- Added CLI tool ``synapse.tools.shutdown`` to facilitate graceful shutdown of
  services by allowing them to complete current tasks.
  (`#4336 <https://github.com/vertexproject/synapse/pull/4336>`_)
- Remove runtsafety requirement for ``init``, ``fini``, and ``empty`` blocks in
  Storm.
  (`#4340 <https://github.com/vertexproject/synapse/pull/4340>`_)
- Added ``synapse.tools.demote`` CLI tool to allow automatic demotion of
  service leaders.
  (`#4341 <https://github.com/vertexproject/synapse/pull/4341>`_)

> - Improved console logging by dropping warning messages for failed
>   normalization properties of a GUID based dictionary object.

Reviewer: ```suggestion
- Improved console logging by dropping warning messages for optional
  properties which fail to normalize in a GUID based dictionary object.
```
